### PR TITLE
Merges up to Bitcoin Core v23 branch point

### PR DESCRIPTION
**Pull request by delta1**
Merges and reviews done so far by myself and @jamesdorfman

### CODE REVIEW

**Comment by jamesdorfman** [comment]
Screenshot 2023-08-29 at 1 58 25 PM

To review all the commits, it appears that you may have to check the PR out locally.

As of now, I have reviewed all of @delta1 commits up until https://github.com/ElementsProject/elements/pull/1262/commits/8dbf577aaa8c7894724def9a4e70e1d4f9b6f54a.

**Comment by delta1** [comment]
CI builds currently failing with the wrong branch name to the new qa-assets repo: 

```diff
diff --git a/ci/test/04_install.sh b/ci/test/04_install.sh
index 7df24e4490..01cb9d1be8 100755
--- a/ci/test/04_install.sh
+++ b/ci/test/04_install.sh
@@ -96,7 +96,7 @@ elif [ "$RUN_UNIT_TESTS" = "true" ] || [ "$RUN_UNIT_TESTS_SEQUENTIAL" = "true" ]
   export DIR_UNIT_TEST_DATA=${DIR_QA_ASSETS}/unit_test_data/
   if [ ! -d "$DIR_UNIT_TEST_DATA" ]; then
     CI_EXEC mkdir -p "$DIR_UNIT_TEST_DATA"
-    CI_EXEC curl --location --fail https://github.com/ElementsProject/qa-assets/raw/main/unit_test_data/script_assets_test.json -o "${DIR_UNIT_TEST_DATA}/script_assets_test.json"
+    CI_EXEC curl --location --fail https://github.com/ElementsProject/qa-assets/raw/master/unit_test_data/script_assets_test.json -o "${DIR_UNIT_TEST_DATA}/script_assets_test.json"
   fi
 fi
 
```

**Comment by delta1** [comment]
pushed 2 commits which should fix the CI builds (except lint)

**Comment by delta1** [comment]
looks like the macOS native CI build might need [53fd8278](https://github.com/bitcoin/bitcoin/commit/53fd827883f9649f091999b59b0d916e0199c796) from bitcoin/bitcoin#26879

**Comment by delta1** [comment]
ACK 5dc85e021435221f16286e016de79b7666198494 for @jamesdorfman commits from me. 

3 out of 4 of the failing CI jobs we know are fixed from upstream backports for 23.x 

Then the TSan CI test will have to be debugged and fixed.

File: src/coins.cpp [hunk 1]
```diff
@@ -113,7 +113,7 @@ void CCoinsViewCache::AddCoin(const COutPoint &outpoint, Coin&& coin, bool possi
            outpoint.hash.data(),
            (uint32_t)outpoint.n,
            (uint32_t)coin.nHeight,
-           (int64_t)coin.out.nValue,
+           coin.out.nValue.IsExplicit() ? (int64_t)coin.out.nValue.GetAmount() : 0,
```
Comment (jamesdorfman) [inline src/coins.cpp:116]: @delta1 it's impossible for an explicit value to be 0, correct?

**Comment by jamesdorfman** [comment]
ACK 5dc85e0. Reviewed all of the commits which were not mine, up until that point.

**Pull request merged**